Libraries/LibWeb/Gamepad/Gamepad.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024, Undefine <undefine@undefine.pl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/Timer.h>
+#include <LibWeb/Bindings/GamepadPrototype.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/Gamepad/Gamepad.h>
+#include <LibWeb/HighResolutionTime/TimeOrigin.h>
+
+namespace Web::Gamepad {
+
+GC_DEFINE_ALLOCATOR(Gamepad);
+
+WebIDL::ExceptionOr<GC::Ref<Gamepad>> Gamepad::create(JS::Realm& realm, NonnullRefPtr<Core::Gamepad> gamepad)
+{
+    return realm.create<Gamepad>(realm, gamepad);
+}
+
+Gamepad::Gamepad(JS::Realm& realm, NonnullRefPtr<Core::Gamepad> gamepad)
+    : Web::Bindings::PlatformObject(realm)
+    , m_gamepad(gamepad)
+{
+    m_poll_timer = Core::Timer::create_repeating(10, [this]() {
+        bool needs_update = m_gamepad->poll_all_events().release_value_but_fixme_should_propagate_errors();
+        if (needs_update)
+            update_gamepad_state();
+    });
+    m_poll_timer->start();
+}
+
+Gamepad::~Gamepad() = default;
+
+void Gamepad::initialize(JS::Realm& realm)
+{
+    Base::initialize(realm);
+    WEB_SET_PROTOTYPE_FOR_INTERFACE(Gamepad);
+}
+
+WebIDL::ExceptionOr<Vector<double>> const Gamepad::axes()
+{
+    // TODO: This not what the spec does
```

Comment:
Please note what you do differently by adding a spec link, copying the spec verbatim and then note it with `// AD-HOC: ...`.

Same goes for `::buttons()`.